internal/liveness/liveness.go

```diff
@@ -37,57 +44,113 @@ var (
 	})
 )
 
-func getLiveness(timeout time.Duration, csiConn *grpc.ClientConn) {
-	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+func (c *probeConn) checkProbe(w http.ResponseWriter, req *http.Request) {
+	ctx, cancel := context.WithTimeout(req.Context(), c.config.ProbeTimeout)
 	defer cancel()
 
-	util.TraceLogMsg("Sending probe request to CSI driver")
-	ready, err := rpc.Probe(ctx, csiConn)
+	util.TraceLog(ctx, "Healthz req: Sending probe request to CSI driver %s", c.config.DriverName)
+	ready, err := rpc.Probe(ctx, c.conn)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		_, err = w.Write([]byte(err.Error()))
+		if err != nil {
+			util.ErrorLog(ctx, "Healthz req: write failed: %v", err)
+		}
+		util.ErrorLog(ctx, "Healthz req: health check failed: %v", err)
+		return
+	}
+
+	if !ready {
+		w.WriteHeader(http.StatusInternalServerError)
+		_, err = w.Write([]byte("Healthz req: driver responded but is not ready"))
+		if err != nil {
+			util.ErrorLog(ctx, "Healthz req: write failed: %v", err)
+		}
+
+		util.ErrorLog(ctx, "Healthz req: driver responded but is not ready")
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	_, err = w.Write([]byte(`ok`))
+	if err != nil {
+		util.ErrorLog(ctx, "Healthz req: write failed: %v", err)
+	}
+	util.ExtendedLog(ctx, "Healthz req: Health check succeeded")
+}
+
+func getLiveness(c *probeConn) {
+	ctx, cancel := context.WithTimeout(context.Background(), c.config.ProbeTimeout)
+	defer cancel()
+
+	util.TraceLog(ctx, "Metrics req: Sending probe request to CSI driver: %s", c.config.DriverName)
+	ready, err := rpc.Probe(ctx, c.conn)
 	if err != nil {
 		liveness.Set(0)
-		util.ErrorLogMsg("health check failed: %v", err)
+		util.ErrorLog(ctx, "Metrics req: health check failed: %v", err)
```

Comment:
ctx logging is not much helpful as it doesn't contain any information on what we need

Reply:
OK. Will, it hurt if we have ctx?

Reply:
it won't hurt, but as we have a separate function for logging without context let's use it.